pkg/workloads/console/runner/runner.go

```diff
@@ -736,10 +769,56 @@ func (c *Runner) GetAttachablePod(csl *workloadsv1alpha1.Console) (*corev1.Pod,
 	}
 
 	for _, c := range pod.Spec.Containers {
-		if c.TTY {
+		if !csl.Spec.Interactive || c.TTY {
```

Comment:
Basically, if it's not an interactive console, then there's going to be no container with TTY = true by definition, so we need another heuristic. In this case, we've just done "take the first one \*shrug\*". It would probably be clearer to have a separate early return. 

I'm not sure if we want a more robust mechanism here for getting 'the right container', mind.